src/utils.ts

```diff
@@ -56,16 +56,26 @@ export function serial<T>(
   );
 }
 
+type Task = ReturnType<typeof console.createTask>;
+const defaultTask = { run: (function_) => function_() } as Task;
+const createTask =
+  typeof console.createTask !== "undefined"
+    ? console.createTask
+    : () => defaultTask;
+
 export function serialCaller(hooks: HookCallback[], arguments_?: any[]) {
+  const task = createTask(this.name);
```

Comment:
Are we using a new `{ name }` context and `this` to avoid breaking changes? I'm wondering if we could introduce `serialTaskCaller` that accepts the task as first argument to avoid this (also little perf/mem saving by not making new context)

Reply:
Yes, that's exactly why. I can try that idea again - the reason I didn't in this PR was (I think) it wrought havoc with the types/caller signature.

Reply:
how about now?